Source/WebKit/UIProcess/WebsiteData/WebsiteDataStore.cpp

```diff
@@ -925,6 +942,11 @@ void WebsiteDataStore::removeData(OptionSet<WebsiteDataType> dataTypes, WallTime
     if (dataTypes.contains(WebsiteDataType::DeviceIdHashSalt) || (dataTypes.contains(WebsiteDataType::Cookies)))
         ensureProtectedDeviceIdHashSaltStorage()->deleteDeviceIdHashSaltOriginsModifiedSince(modifiedSince, [callbackAggregator] { });
 
+#if ENABLE(ENCRYPTED_MEDIA)
+    if (dataTypes.contains(WebsiteDataType::MediaKeys) || (dataTypes.contains(WebsiteDataType::Cookies)))
+        ensureProtectedMediaKeysHashSaltStorage()->deleteDeviceIdHashSaltOriginsModifiedSince(modifiedSince, [callbackAggregator] { });
+#endif
+
```

Comment:
Nit: why wrap this with `if ENABLE(ENCRYPTED_MEDIA)` when the block below does not?

Reply:
Ideally, because we shouldn't be creating directories for settings that can never get accessed (because they're disabled at compile time).